webui/react/src/hooks/useModal/Workspace/useModalWorkspaceCreate.tsx

```diff
@@ -215,7 +216,13 @@ const useModalWorkspaceCreate = ({ onClose, workspaceID }: Props = {}): ModalHoo
           useAgentGroup,
           checkpointStorageConfig,
         } = values;
-        const body = {
+        const body: {
+          agentUserGroup?: V1AgentUserGroup;
+          checkpointStorageConfig?: unknown;
+          name: string;
+        } = {
+          agentUserGroup: {},
+          checkpointStorageConfig: undefined,
```

Comment:
Be careful when doing this, sending `agentUserGroup: {}` would remove existing agentUserGroup if there is any (with relevant permission), and sending payload without `agentUserGroup` will not trigger this change

Reply:
what is this supposed to be?

Reply:
Only include `agentUserGroup` and  `checkpointStorageConfig` if the user has relevant permissions

Reply:
ok updated. could you check it again?